### DISCUSSION TITLE
Retrieving the referer with query params (Next 14 with app dir)

### DISCUSSION THREAD

**Discussion by MickaelMesnage** [vercel/next.js#64197] [App Router]
Hello,

I have a first site [https://www.something.com?toto=tata](https://www.something.com/?toto=tata) that calls my Next.js 14with app directory website. I would like to retrieve the full referer including the query params.
In my server component, I have implemented the following code, but I only retrieve the pathname without the query params :

```
import { headers } from "next/headers";

export default async function Page() {
  const headersList = headers();
  headersList.get()
  const referer = headersList.get("referer");

  console.log({ referer });
  return <span>Ref</span>;
}
```

Do you have a solution?
Thank you in advance,
Mickaël

**Comment by pcintra11** [comment]
I have the same problem
